show line number for each result text

BarretRen opened the pull request:
Hi, this plugin is so wonderful and I use it everyday. And I found an bug (if we can call it a bug) for the line number, the pull request is used to fix it.

During the search, if there are more than one result found in the same file, we can only see the the line/column numbers of the first matched result showing behand the filename. 
Here is an examle:
![image](https://github.com/nvim-pack/nvim-spectre/assets/13688920/22dfae42-79b2-46f6-8e17-a1eaba42ef4b)

I think it is useless, because we still can't know the line number of each matched result.
The following formats is more useful. we can see which file the results belong to, and we can also know the line number of each result. 
![image](https://github.com/nvim-pack/nvim-spectre/assets/13688920/a2957354-c377-4063-b560-c55b5566d76d)

This pull request is used to show the new formats, please help to check it.

BarretRen commented:
After such a long time, nothing happens. So, close this. Fuck that.

EpsilonKu commented:
![Screenshot from 2024-02-05 16 53 30](https://github.com/nvim-pack/nvim-spectre/assets/63652620/2e1a93fc-1bfd-45a2-bf12-fbc506265fa6)
Well, the highlight is moved to left.
https://github.com/nvim-pack/nvim-spectre/blob/817944747413ead7438ec28334201a340898b7e6/lua/spectre/ui.lua#L31-L33C54

EpsilonKu commented:
If you busy, I can add myself. Sorry, for late reply. I was busy.

BarretRen commented:
Thanks for reopen this pull request and very sorry about my rude words before. I have rebased the changes with the master branch and added a new changeset to fix the highlight issue. 
It seems ok on my server, here is the test result:
![image](https://github.com/nvim-pack/nvim-spectre/assets/13688920/6bb044ca-7035-422d-9d31-54a99f2a5208)

I didn't notice the highlight before and forget to check this. Sorry about that and sorry about my rude words in previous comment again.

EpsilonKu commented:
There is another 3 call in `init.lua` which calls `render_line`. Can you add them also `item_line`?
I got error when typed replace word because it calls another `render_line`.

EpsilonKu commented:
> Sorry about that and sorry about my rude words in previous comment again.
It helped me finally look for this project again. Thank you, lol.

BarretRen commented:
> There is another 3 call in `init.lua` which calls `render_line`. Can you add them also `item_line`? I got error when typed replace word because it calls another `render_line`.
new changeset is pushed, please help to check it.

EpsilonKu commented:
Well, it fails because text doesn't match up because of number row. Can you add option to disable it and enable it by default. I will disable on tests later. So if it done, pretty much ready for MR.

BarretRen commented:
new option lnum_for_results is added:
```lua
lnum_for_results   = true, -- show line number for search/replace results
```

EpsilonKu commented:
LGTM
